lib/node-token-validation/src/api/TokenValidator.ts

```diff
@@ -49,6 +50,136 @@ export class TokenValidator {
+                        next(error);
+                    }
+    
+                    // Attach bearer token to authorization header
+                    req.headers.authorization = `Bearer ${token}`;
+                }
+            }
+
+            // Validating token from request
+            this.validateTokenFromRequest(req, options)
+                .then(() => {
+                    next();
+                })
+                .catch((error) => {
+                    next(error);
+                });
+        };
+    }
+
+    /**
+     * Function to validate token from request
+     * @param request 
+     * @param {@link (TokenValidationParameters:type)}
+     * @returns {Promise.<TokenValidationResponse>}
+     */
+    async validateTokenFromRequest(request: ExpressRequest, options: TokenValidationParameters): Promise<TokenValidationResponse> {
+        this.logger.trace("TokenValidator.validateTokenFromRequest called");
+
+        // Validates token if found in authorization header
+        if (request.headers && request.headers.authorization) {
+            const authComponents = request.headers.authorization.split(" ");
+            
+            // Validates token if authorization header is bearer
+            if (authComponents.length === 2 && authComponents[0].toLowerCase() === AuthenticationScheme.BEARER.toLowerCase()) {
+                const token: string = authComponents[1];
+                this.logger.verbose("Bearer token extracted from request authorization headers");
+                return this.validateToken(token, options);
+            } else {
+                this.logger.verbose("Request authorization headers does not include bearer token");
+                // If not bearer, call CAE/EasyAuth/MISE solution here for more complex handling
+            }
+        } 
+        
+        // Validates token if found in request body
+        if (request.body && request.body.access_token) {
+            this.logger.verbose("Token extracted from request body");
+            return this.validateToken(request.body.access_token, options);
+        }
+
+        throw ValidationConfigurationError.createMissingTokenError("No tokens found in authorization header or body.");
+    }
+
+    /**
+     * Function to validate token from msal-node token acquisition response. Returns array of token validation responses.
+     * @param response 
+     * @param {@link (TokenValidationParameters:type)}
+     * @returns {Promise.<TokenValidationResponse>}
+     */
+    async validateTokenFromResponse(response: AuthenticationResult, idTokenOptions?: TokenValidationParameters, accessTokenOptions?: TokenValidationParameters): Promise<TokenValidationResponse[]> {
+        this.logger.trace("TokenValidator.validateTokenFromResponse called");
+
+        // Only validates token if tokenType from response is Bearer
+        if (response.tokenType === AuthenticationScheme.BEARER) {
```

Comment:
Do we need to force casing here?

Reply:
As opposed to case-insensitive?

Reply:
Ah, I changed it to match the other bearer comparison earlier in the code. Let me know if that's not what you meant.